web_poet/rules.py

```diff
@@ -0,0 +1,352 @@
+        instead_of: Optional[Type[ItemPage]] = None,
+        to_return: Optional[Type] = None,
+        exclude: Optional[Strings] = None,
+        priority: int = 500,
+        **kwargs,
+    ):
+        """
+        Class decorator that indicates that the decorated Page Object should work
+        for the given URL patterns.
+
+        The URL patterns are matched using the ``include`` and ``exclude``
+        parameters while ``priority`` breaks any ties. See the documentation
+        of the `url-matcher <https://url-matcher.readthedocs.io/>`_ package for
+        more information about them.
+
+        This decorator is able to derive the Item Class returned by the Page
+        Object (see :ref:`item-class-example` section for some examples). This is
+        important since it marks what type of item the Page Object is capable of
+        returning for the given URL patterns. For certain advanced cases, you can
+        pass a ``to_return`` parameter which replaces any derived values (though
+        this isn't generally recommended).
+
+        Passing another Page Object into the ``instead_of`` parameter indicates
+        that the decorated Page Object will be used instead of that for the given
+        set of URL patterns. This is the concept of **overrides** (see the
+        :ref:`intro-overrides` section for more info`).
+
+        Any extra parameters are stored as meta information that can be later used.
+
+        :param include: The URLs that should be handled by the decorated Page Object.
+        :param instead_of: The Page Object that should be `replaced`.
+        :param to_return: The Item Class holding the data returned by the Page Object.
+            This could be omitted as it could be derived from the ``Returns[ItemClass]``
+            or ``ItemPage[ItemClass]`` declaration of the Page Object. See
+            :ref:`item-classes` section. Code example in :ref:`combination` subsection.
+        :param exclude: The URLs over which the override should **not** happen.
+        :param priority: The resolution priority in case of `conflicting` rules.
+            A conflict happens when the ``include``, ``override``, and ``exclude``
+            parameters are the same. If so, the `highest priority` will be
+            chosen.
+        """
+
+        def wrapper(cls):
+
+            if overrides is not None:
+                msg = (
+                    "The 'overrides' parameter in @handle_urls is deprecated. "
+                    "Use the 'instead_of' parameter."
+                )
+                warnings.warn(msg, DeprecationWarning, stacklevel=2)
+
+            rule = ApplyRule(
+                for_patterns=Patterns(
+                    include=as_list(include),
+                    exclude=as_list(exclude),
+                    priority=priority,
+                ),
+                use=cls,
+                instead_of=instead_of or overrides,
+                to_return=to_return or get_item_cls(cls),
```

Comment:
Would this make sense?

```suggestion
                to_return=to_return or get_item_cls(cls) or get_item_cls(instead_of or overrides),
```